agent/utils/compare_versions.go

```diff
@@ -40,7 +43,8 @@ type semver struct {
 // * <x.y.z -- Matches a version less than the selector version
 // * x.y.z,a.b.c -- Matches if the version matches either of the two selector versions
 func (lhs Version) Matches(selector string) (bool, error) {
-	lhsVersion, err := parseSemver(string(lhs))
+	lhsVer := appendIfNoPatch(string(lhs))
```

Comment:
Let's not do `appendIfNoPatch` anywhere in this method. Instead, do that in `getDockerClientForVersion` so that you don't have to keep doing that at multiple places here. Let's minimize changes to `Version` type as much as possible.

Reply:
If there are asynchronous failures for containers because of a specific versioned docker client  not being there, having these errors printed out during discovery helps us debug the issue. If we make these `Debug`, we'll lose these messages when logging at default `Info` level.